src/coprocessor/endpoint.rs

```diff
@@ -645,32 +643,28 @@ impl TiDbEndPoint {
         }
     }
 
-    fn handle_select(&self, sel: SelectRequest, t: &mut RequestTask) -> Result<Response> {
-        let ctx = SelectContext::new(sel, self.snap.as_ref(), &mut t.statistics, &t.ctx)?;
-        let range = t.req.get_ranges().to_vec();
-        ctx.handle_request(range)
+    fn handle_select(self, sel: SelectRequest, t: &mut RequestTask) -> Result<Response> {
+        let mut ctx = SelectContext::new(sel, self.snap, t.ctx.clone())?;
+        let ranges = t.req.take_ranges().into_vec();
+        let res = ctx.handle_request(ranges)?;
+        ctx.collect_statistics_into(&mut t.statistics);
```

Comment:
Should we still collect statistics when any error happens?

Reply:
I think so. @BusyJay what do you think?

Reply:
Agree.